src/script/interpreter.cpp

```diff
@@ -1167,6 +1199,14 @@ uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsig
             hashOutputs = ss.GetHash();
         }
 
+        if (!txTo.sapData->vShieldedSpend.empty()) {
```

Comment:
now we know that `sapData` is always initialized... but  in the future this might be trying to access a `nullopt`

Reply:
done